src/lxc/commands.c

```diff
@@ -180,15 +182,14 @@ static int lxc_cmd_rsp_recv(int sock, struct lxc_cmd_rr *cmd)
 		errno = ENOMEM;
 		SYSERROR("Failed to allocate response buffer for command \"%s\"",
 		         lxc_cmd_str(cmd->req.cmd));
-		return -ENOMEM;
+		return -1;
 	}
 
-	ret = recv(sock, rsp->data, rsp->datalen, 0);
+	ret = lxc_recv_nointr(sock, rsp->data, rsp->datalen, 0);
 	if (ret != rsp->datalen) {
 		SYSERROR("Failed to receive response data for command \"%s\"",
 		         lxc_cmd_str(cmd->req.cmd));
-		if (ret >= 0)
-			ret = -1;
+		return -1;
```

Comment:
The callers seem to want to handle `ECONNRESET`, which AFAIK can occur here as well. Previously we only had the case above where we explicitly returned `-ECONNRESET`, so this looks like it was an actual issue here.